### DISCUSSION TITLE
SC-126 Running v3.0.1 - FLASH.COM - Unknown Flash Chip 0xC340

### DISCUSSION THREAD

**Discussion by jjandersoj** [wwarthen/RomWBW#385] [Q&A] [answered]
Trying to upgrade my older SC-126.  I'm getting:

FLASH4 by Will Sowerbutts <will@sowerbutts.com> version 1.2.3

Using RomWBW (v2.6+) bank switching.
Flash memory chip ID is 0xC340: Unknown flash chip.
Your flash memory chip is not recognised.
Please email will@sowerbutts.com if you would like support for your
system added to this program.

I bought the system used/assembled,

I don't seem to have the XMODEM loader choice in the boot menu.

I'd like to get to v3.4.0 RomWBW,

Thanks  John-A

**Comment by wwarthen** [comment, marked as answer]
It is probably due to the ROM write protect jumper being set.  In that case, the ROM chip cannot be identified.  Remove the ROM write protect jumper and try again.  The SC126 jumpers are documented at [Small Computer Central](https://smallcomputercentral.com/sc126-z180-motherboard-rc2014/).  Jumpers J1 and J2 are the specific jumpers.  The specific jumper depends on which ROM position you are using for RomWBW.

Thanks,

Wayne

**Reply by jjandersoj** [reply]
Thanks Wayne.  (Doing my best Homer Simpson forehead slap… DOH!)

**Reply by wwarthen** [reply]
Well, you are in good company.  Many people have gotten caught by this.  It is not at all intuitive that recognizing a ROM would require that it be "writable".  I am going to review the applicable RomWBW manuals and make sure this is highlighted.

Thanks,

Wayne

**Comment by jjandersoj** [comment]
That worked beautifully.  I'm officially on 3.4!  For my next question, is there a way to set a fully autmatice boot.  Similar to - Boot 4.3 on a reset?  Basically a full turnkey boot to the OS.  I've been humting in the docs and haven't found any mention,  
JA

**Reply by wwarthen** [reply]
Hi @jjandersoj,

Yes, a fully automatic boot is possible.

The best and easiest way is to use a front panel.  However, I assume you do not have one.

So, you will need to build a custom ROM.  The RomWBW User Guide touches on this in section 12.2 ROM Customization.  However, the bulk of the documentation on building a custom ROM is in the ReadMe.txt file in the Source folder of the distribution.

To implement a fully automatic boot as you described, you will want to modify a couple of configuration settings in the `SCZ180_sc126.asm` configuration file.

This line near the top defines what will be done by default:

```
#DEFINE	BOOT_DEFAULT	"H"		; DEFAULT BOOT LOADER CMD ON <CR> OR AUTO BOOT
```

As you can see, it will normally just execute an "H" command which displays help.  You can change this to any command you want.  So, in your case, you probably want it to be:

```
#DEFINE	BOOT_DEFAULT	"4.3"		; DEFAULT BOOT LOADER CMD ON <CR> OR AUTO BOOT
```

Additionally, you will need to set a timeout.  This is just how long the system will pause before automatically executing the default command.  It is normally set to -1 which means no timeout (wait forever and never execute the default command).  You can set the value to 0 which would mean execute the default command immediately.  However, I would suggest using something like 5 seconds instead.  This would allow you to intercept the automatic boot as needed.  If you start typing before the timeout expires, then the timeout is aborted.  The timeout setting is not included in the standard config file.  You will need to add a line like this:

```
BOOT_TIMEOUT	.EQU	5		; AUTO BOOT TIMEOUT IN SECONDS, -1 TO DISABLE, 0 FOR IMMEDIATE
```

Thanks,

Wayne
